db/multi_cf_iterator_impl.cc

```diff
@@ -0,0 +1,70 @@
+//  Copyright (c) Meta Platforms, Inc. and affiliates.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "db/multi_cf_iterator_impl.h"
+
+#include <cassert>
+
+namespace ROCKSDB_NAMESPACE {
+
+void MultiCfIteratorImpl::SeekToFirst() {
+  Reset();
+  assert(cfhs_.size() == iterators_.size());
+  int i = 0;
+  for (auto& iter : iterators_) {
+    auto& cfh = cfhs_[i];
+    iter->SeekToFirst();
+    if (iter->Valid()) {
+      assert(iter->status().ok());
+      min_heap_.push(MultiCfIteratorInfo{iter, cfh, i});
+    } else {
+      considerStatus(iter->status());
+    }
+    ++i;
+  }
+}
+
+void MultiCfIteratorImpl::Next() {
+  assert(Valid());
+
+  auto* current = min_heap_.top().iterator;
+  std::string current_key_copy =
+      std::string(current->key().data(), current->key().size());
```

Comment:
Call current->Next() once before while loop?

Reply:
if `current->Next()` is called before the loop, the key will be different from the previous key, `comparator_->Compare(current->key(), current_key_copy)` will always return non-zero and we won't get into while loop.

Essentially this `Next()` does the following (maybe I should add this as comment at the top of the function)
1. Get the current key at the top of the heap iterator and remember it
2. Move to the next item in the top iterator
3. if the iterator is still valid, re-heapify. Otherwise, pop it from the heap (which also re-heapify). This may bring the same key that we've seen in 1 to the top of the heap. (Just stored in a different CF)
4. Compare the new top key with the key in 1. If the same, go back to step 2.
5. If the heap is empty, we are done with the iteration.